src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestGetCallGen.generated.txt

```diff
@@ -85,17 +93,25 @@ internal static class GeneratedConfigurationBinder
         string key;
         foreach (Microsoft.Extensions.Configuration.IConfigurationSection section in configuration.GetChildren())
         {
-            if (section.Key is string stringValue6)
+            if (section.Key is string stringValue8)
             {
-                key = stringValue6;
+                key = stringValue8;
                 string element;
-                if (section.Value is string stringValue7)
+                if (section.Value is string stringValue9)
                 {
-                    element = stringValue7;
+                    element = stringValue9;
                     obj[key] = element;
                 }
             }
         }
     }
 
+    public static bool HasChildren(global::Microsoft.Extensions.Configuration.IConfiguration configuration)
```

Comment:
An alternative might be to keep the children around in the generated code and pass them down, since they'll be needed by the BindCore call.  Maybe a struct implementation of IConfiguration (or a subset of it) that is just an immutable snapshot of a configuration node 🤔

Reply:
Interesting idea. Might tinker with this depending on where we go with the API proposal.